ClientApp/KachnaOnline/src/app/models/events/event-modification.model.ts

```diff
@@ -0,0 +1,20 @@
+// event-modification.model.ts
+// Author: David Chocholatý
+
+import { ClubState } from "../states/club-state.model";
+
+export class EventModification {
```

Comment:
How is this different from `Event`? I don't really see a difference hence I would use the same class as was used previously. If there is a difference only in a field or two, I'd use inheritance.

Reply:
There are `string`s for date instead of `Date` type. That is necessary to send correct time (hours, exactly) to the BE. We are using this method for both events and states. When date time variables are handled properly, considering timezones, both these models can be merged into one.

Reply:
Have you tried using `string | Date` as the data type for `from` and `to`? Or would it result in too many needed type checks across the existing code?

Reply:
I have not, but that would be really unpleasant to work with. Check the current solution with usage of inheritance. That makes the code much more maintainable and cleaner, and needs no additional type checks. I would stick with the inheritance solution.